File: PeachPie Extension Libraries/Peachpie.Base.Ext/src/Reactive/ReactiveObject.cs
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace Php.Reactive;

public class ReactiveObject : INotifyPropertyChanged
{
    public event PropertyChangedEventHandler? PropertyChanged;
    
    protected virtual void OnPropertyChanged([CallerMemberName] string? propertyName = null)
    {
        PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
    }
}

File: PeachPie Extension Libraries/Peachpie.Base.Ext/.sdk/Php/Reactive/ReactiveObject.php
<?php

namespace Php\Reactive;

use System\ComponentModel\PropertyChangedEventArgs;
use System\ComponentModel\PropertyChangedEventHandler;

class ReactiveObject
{
    /**
     * @var PropertyChangedEventHandler
     */
    public PropertyChangedEventHandler $PropertyChanged;

    /**
     * @param string|null $propertyName
     * @return void
     */
    protected function onPropertyChanged(string $propertyName = null): void
    {
        $this->PropertyChanged->Invoke($this, new PropertyChangedEventArgs($propertyName));
    }
}